### PR TITLE
Handle empty json response

### PR DESCRIPTION
**Pull request by airjp73**
Addresses #89.

Of course, I'm happy to fix anything if it's not quite right 👍 .

### CODE REVIEW

File: src/fetchBaseQuery.ts [hunk 1]
```diff
@@ -2,18 +2,35 @@ import { joinUrls } from './utils';
...
+    return responseHandler(response);
+  }
+
+  if (responseHandler === 'text') {
+    return response.text();
+  }
+
+  if (responseHandler === 'json') {
+    const text = await response.text();
+    return text.length ? JSON.parse(text) : null;
```
Comment (phryneas) [inline src/fetchBaseQuery.ts:30]: Hmm.
Not sure what this should be.
I'm tending more towards `undefined`.
@msutkowski @markerikson can you take a look?

Comment (airjp73) [reply src/fetchBaseQuery.ts:30]: Yeah, I can definitely see the case for this being `undefined` instead. Happy to update once the others weigh in.

Comment (msutkowski) [reply src/fetchBaseQuery.ts:30]: undefined is probably better... the body was technically undefined in this case, ya?

File: test/fetchBaseQuery.test.tsx [hunk 2]
```diff
@@ -89,6 +89,22 @@ describe('fetchBaseQuery', () => {
       expect(res.data).toEqual({ value: 'success' });
     });
 
+    it('should return null for a simple GET request when the response is null', async () => {
```
Comment (phryneas) [inline test/fetchBaseQuery.test.tsx:92]:
```suggestion
    it('should return undefined for a simple GET request when the response is empty', async () => {
```

File: test/mocks/server.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { rest } from 'msw';
 export const server = setupServer(
   rest.get('http://example.com/success', (_, res, ctx) => res(ctx.json({ value: 'success' }))),
   rest.post('http://example.com/success', (_, res, ctx) => res(ctx.json({ value: 'success' }))),
+  rest.get('http://example.com/empty', (_, res, ctx) => res(ctx.json(null))),
```
Comment (phryneas) [inline test/mocks/server.ts:10]:
```suggestion
  rest.get('http://example.com/empty', (_, res, ctx) => res(ctx.body(''))),
```

**Comment by phryneas** [comment]
Let's get this merged.

**Pull request merged**